### ISSUE TITLE
unnecessary_lazy_evaluations help produces code which formatted differently than format

### ISSUE THREAD

**Issue by Fishrock123** [rust-lang/rust-clippy#8394]
### Summary

A clippy lint produced a hint which is different than how rustfmt formats the code, which, seems weird?

### Reproducer

I tried this code:

```rust
    // Get the name. This is non-optional.
    let name = parts
        .next()
        .ok_or(HeaderError::ServerTimingInvalid("must include a name"))?
        .trim_end();

```

(Note that it is already one level indented)

I expected to see this happen:

```
warning: unnecessary closure used to substitute value for `Option::None`
  --> src/trace/server_timing/parse.rs:33:16
   |
33 |       let name = parts
   |  ________________^
34 | |         .next()
35 | |         .ok_or_else(|| HeaderError::ServerTimingInvalid("must include a name"))?
   | |_______________________________________________________________________________^
   |
   = note: `#[warn(clippy::unnecessary_lazy_evaluations)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `ok_or` instead
   |
33 ~     let name = parts
34 ~         .next()
35 ~         .ok_or(HeaderError::ServerTimingInvalid("must include a name"))?
   |
```

Instead, this happened:

```
warning: unnecessary closure used to substitute value for `Option::None`
  --> src/trace/server_timing/parse.rs:33:16
   |
33 |       let name = parts
   |  ________________^
34 | |         .next()
35 | |         .ok_or_else(|| HeaderError::ServerTimingInvalid("must include a name"))?
   | |_______________________________________________________________________________^
   |
   = note: `#[warn(clippy::unnecessary_lazy_evaluations)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `ok_or` instead
   |
33 ~     let name = parts
34 ~         .next().ok_or(HeaderError::ServerTimingInvalid("must include a name"))?
   |
```

### Version

```text
rustc 1.58.1 (db9d1b20b 2022-01-20)
binary: rustc
commit-hash: db9d1b20bba1968c1ec1fc49616d4742c1725b4b
commit-date: 2022-01-20
host: x86_64-unknown-linux-gnu
release: 1.58.1
LLVM version: 13.0.0
```

**Comment by matthiaskrgr** [comment]
It's not in the scope of the clippy/rustfix project to produce rustfmt'ed code.
It's kind of impossible for clippy to do that since clippy has no idea what your `rustfmt.toml` configuration looks like.

Just  run `cargo fmt` after running `cargo clippy --fix` :slightly_smiling_face:
